### PR TITLE
ci(Gha): Migrate deploy next workflow 

### LINKED ISSUE #1299: Migrate build test & deploy next to GHA
To migrate the current circleci build, test & deploy next workflow to GitHub actions.

### PR DESCRIPTION
**Pull request by elglup**
## Related issue

closes #1299 

## Overview

Migrate build test and deploy next workflow to GitHub actions

## Work carried out

- [x] Created build, test and deploy new GitHub actions workflow
- [x] Disabled build, test and deploy workflow in circle ci config

### CODE REVIEW

**Comment by elglup** [comment]
Need to add design-tokens

**Comment by elglup** [comment]
> Need to add design-tokens

Done

File: .github/workflows/build_test_and_deploy_next.yml [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        - name: Build design-tokens
+          run: yarn --cwd packages/design-tokens build
+
+        - name: Create tar
+          run: |
+            cp -R packages/icon-library/dist distil
+            cp -R packages/design-tokens/dist distdt
+            tar -czvf dist.tar.gz distil distdt
+
+        - name: Persist icon-library
```
Comment (m7kvqbe1) [inline .github/workflows/build_test_and_deploy_next.yml]: Maybe we should rename this to 'Persist artefacts' or something?

**Review by m7kvqbe1 — CHANGES_REQUESTED** [review]
LGTM - just left a comment about the name of a step in the Build_prerequisites job

File: .github/workflows/build_test_and_deploy_next.yml (hunk 1 above)
Comment (elglup) [reply .github/workflows/build_test_and_deploy_next.yml]: Fair point as its now doing more than just the icon-library, I'll rename.

Comment (elglup) [reply .github/workflows/build_test_and_deploy_next.yml]: Renamed step, also did the same for build_and_test_feature.

**Review by m7kvqbe1 — APPROVED** [review]
👍

**Pull request merged**